### PR TITLE
Download metadata json files on start up

### PR DESCRIPTION
**Pull request by zachthompson**
Attempt to download metdata json files from https://duck.co on start before resorting to using previously downloaded json.

### CODE REVIEW

**Comment by russellholt** [comment]
Once we get the final word on the temp / download location, should be good to go

**Comment by zachthompson** [comment]
@russellholt storage location finalized.

**Comment by russellholt** [comment]
:+1:

File: lib/DDG/Meta/Data.pm [hunk 1]
```diff
@@ -25,25 +26,44 @@ unless(%ia_metadata){
...
+        if($c == RC_OK || $c == RC_NOT_MODIFIED){
+            $json = io($f)->slurp; 
+        }
+        else {
+            debug && warn "Failed to download metdata for $iat: $c";
+            if(-f $f){
+                $json = io($f)->all;
+            }
+            else{
+               warn "Failed to download metadata for $iat and no local file $f";
```
Comment (russellholt) [inline lib/DDG/Meta/Data.pm]: fatal error here?
fallback on the dist file metadata?
```perl
eval{ dist_file("DDG-${iat}Bundle-OpenSourceDuckDuckGo", lc $iat . '/meta/metadata.json') }
```

Comment (zachthompson) [inline lib/DDG/Meta/Data.pm]: I think we want to depend on /tmp as the fallback.  Whether it's a fatal error or not is up for debate, e.g. when only running spices do we really need the Longtail json?

**Comment by russellholt** [comment]
LGTM. questions answered.

**Pull request merged**